### DISCUSSION TITLE
Listing all source:layer ID pairs in a PMTiles file from CLI?

### DISCUSSION THREAD

**Discussion by diehl** [protomaps/PMTiles#416] [Q&A] [answered]
I'm still struggling to figure out how I can apply existing OpenMapTiles styles to PMTiles constructed with tilemaker. There are often mismatches, some of which are not immediately obvious. Is there a mechanism by which I can dump a list of all the (source, layer ID) pairs that occur in the PMTiles file? I'm hoping I could use such a list to then discern programmatically where the mismatches are when compared to a given JSON style specification. I'm guessing the PMTiles viewer does this implicitly to render all the available layers in a given PMTiles file.

**Comment by bdon** [comment, marked as answer]
It's in the `metadata` under `vector_layers`:

`
pmtiles show https://build.protomaps.com/20240628.pmtiles --tilejson
`

The PMTiles viewer uses `vector_layers`.

I don't think it will be easy to transfer an OMT style to a non-OMT tileset, you should probably just generate a OMT tileset with Planetiler.

**Reply by diehl** [reply]
Thanks @bdon - I've only explored Tilemaker and Openmaptiles so far. If I can build tiles with Planetiler that work straightaway with OMT styles, that'd be a huge win. If memory serves, don't you have a prebuilt OMT tileset for the planet available for download? I have a vague memory of seeing that somewhere. If not, I'll dig in to Planetiler to build a map for the planet and learn how to deploy that with Cloudflare.

**Reply by diehl** [reply]
Scratch that. Just found the downloads section...
